File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> v1;
        vector<int> v2;
        vector<int> v3;
        vector<int> concatenated;
        int count = 0;
        for(int i=0;i<nums.size();i++){
            if(nums[i] < pivot){
                v1.push_back(nums[i]);
            }else if(nums[i] == pivot){
                v2.push_back(nums[i]);
            }else{
                v3.push_back(nums[i]);
            }
        }

     concatenated.reserve(v1.size() + v2.size() + v3.size());

    // Copy elements using std::copy
    std::copy(v1.begin(), v1.end(), std::back_inserter(concatenated));
    std::copy(v2.begin(), v2.end(), std::back_inserter(concatenated));
    std::copy(v3.begin(), v3.end(), std::back_inserter(concatenated));
       
        return concatenated;
    }
};

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        // Lists to store numbers in three categories
        List<Integer> less = new ArrayList<>();
        List<Integer> equal = new ArrayList<>();
        List<Integer> greater = new ArrayList<>();

        // Categorize numbers
        for (int num : nums) {
            if (num < pivot) {
                less.add(num);
            } else if (num == pivot) {
                equal.add(num);
            } else {
                greater.add(num);
            }
        }

        // Merging lists into a single array
        int[] result = new int[nums.length];
        int index = 0;

        for (int num : less) result[index++] = num;
        for (int num : equal) result[index++] = num;
        for (int num : greater) result[index++] = num;

        return result;
    }
}